### PR TITLE
Arcconf Plugin: Add 'pool-member-info' support

### PR DESCRIPTION
**Pull request by sharath-t**
implementation for 'pool-member-info' API support

### CODE REVIEW

File: plugin/arcconf/arcconf.py [hunk 1]
```diff
@@ -644,3 +644,98 @@ def volume_delete(self, volume, flags=0):
...
+        Depends on command:
+            arcconf getconfigjson <ctrlNo>
+        """
+
+        if not pool.plugin_data:
+            raise LsmError(
+                ErrorNumber.INVALID_ARGUMENT,
+                "Illegal input pool argument: missing plugin_data property")
+
+        pool_info = pool.plugin_data
```
Comment (cathay4t) [inline plugin/arcconf/arcconf.py]: `plugin_data` should be a string.

File: plugin/arcconf/arcconf.py [hunk 2]
```diff
@@ -644,3 +644,98 @@ def volume_delete(self, volume, flags=0):
...
+        ctrl_id = pool_info['ctrl_id']
+        array_id = pool_info['array_id']
+        consumer_array_id = None
+        raid_level = Volume.RAID_TYPE_UNKNOWN
```
Comment (cathay4t) [inline plugin/arcconf/arcconf.py]: The `raid_level` here is the index number of `_ARCCONF_RAID_LEVEL_CONV` which has nothing to do with `Volume.RAID_TYPE_UNKNOWN` constant.
And worse, the `Volume.RAID_TYPE_UNKNOWN` is -1 which give you `Volume.RAID_TYPE_RAID60` when looking up `_ARCCONF_RAID_LEVEL_CONV`

The proper way to do arcconf raid level to lsm raid level converting is using `_arcconf_raid_level_to_lsm()`.

File: plugin/arcconf/arcconf.py [hunk 3]
```diff
@@ -644,3 +644,98 @@ def volume_delete(self, volume, flags=0):
...
+
+        ctrl_info = self._arcconf_exec(['GETCONFIGJSON', ctrl_id])
+        ctrl_json_info = self._filter_cmd_output(ctrl_info)
+        device_info = ctrl_json_info['Controller']['Channel']
+        volume_info = ctrl_json_info['Controller']['LogicalDrive']
+
+        for volume in volume_info:
+            chunk_data = volume['Chunk']
+            for chunk in chunk_data:
+                consumer_array_id = chunk['consumerArrayID']  # consumerArrayID in all the chunk will be same
```
Comment (cathay4t) [inline plugin/arcconf/arcconf.py]: The text width should be less than 80.

File: plugin/arcconf/arcconf.py [hunk 4]
```diff
@@ -644,3 +644,98 @@ def volume_delete(self, volume, flags=0):
...
+            if consumer_array_id == array_id:
+                raid_level = volume['raidLevel']
+
+        for device in device_info:
+            if 'HardDrive'in device.keys():
+                for hard_drive in device['HardDrive']:
+                    chunk_data = hard_drive['Chunk']
+                    for chunk in chunk_data:
+                        if ('consumerArrayID' in chunk.keys()) and (chunk['consumerArrayID'] == array_id):
```
Comment (cathay4t) [inline plugin/arcconf/arcconf.py]: The text width should be less than 80.

File: plugin/arcconf/arcconf.py (hunk 1 above)
Comment (sharath-t) [reply plugin/arcconf/arcconf.py]: done - commit id - 9a67e0bbaf53081e2da601513d85b0d33b767ba8

File: plugin/arcconf/arcconf.py (hunk 2 above)
Comment (sharath-t) [reply plugin/arcconf/arcconf.py]: thanks for the catch.

done - commit id - 9a67e0bbaf53081e2da601513d85b0d33b767ba8

File: plugin/arcconf/arcconf.py (hunk 3 above)
Comment (sharath-t) [reply plugin/arcconf/arcconf.py]: done - commit id - 9a67e0bbaf53081e2da601513d85b0d33b767ba8

File: plugin/arcconf/arcconf.py (hunk 4 above)
Comment (sharath-t) [reply plugin/arcconf/arcconf.py]: done - commit id - 9a67e0bbaf53081e2da601513d85b0d33b767ba8

**Comment by tasleson** [comment]
Integrated into https://github.com/libstorage/libstoragemgmt/pull/386

**Pull request closed without merging**